okhttp/src/main/kotlin/okhttp3/Address.kt

```diff
@@ -59,6 +61,44 @@ class Address(
    */
   @get:JvmName("proxySelector") val proxySelector: ProxySelector,
 ) {
+  /** Returns the socket factory for new connections. */
+  @get:JvmName("socketFactory")
+  val socketFactory: SocketFactory
+    get() = (okioSocketFactory as RealOkioSocketFactory).delegate
```

Comment:
I think this hurts some of the benefits of having simpler decoration, delegation of an interface.  Should we have clean ways to unwrap and wrap without knowing the concrete type?

```
interface OkioSocketFactory {
...
  val platformSocketFactory: SocketFactory?
...
}

interface OkioSslSocketFactory: OkioSocketFactory {
...
  override val platformSocketFactory: SslSocketFactory?
...
}
```

Reply:
I agree it should be possible, but I think a way to break out when the native implementation has features you need without assuming Real* will come in handy.

And we demonstrate we want it here in a few places.